sljit_src/sljitNativeX86_common.c

```diff
@@ -557,7 +573,8 @@ SLJIT_API_FUNC_ATTRIBUTE void* sljit_generate_code(struct sljit_compiler *compil
 					SLJIT_ASSERT(put_label->label);
 					put_label->addr = (sljit_uw)code_ptr;
 #if (defined SLJIT_CONFIG_X86_64 && SLJIT_CONFIG_X86_64)
-					code_ptr = generate_put_label_code(put_label, code_ptr, (sljit_uw)(SLJIT_ADD_EXEC_OFFSET(code, executable_offset) + put_label->label->size));
+					SLJIT_ASSERT(UWORD_MAX - (sljit_uw)code >= put_label->label->size);
```

Comment:
`(sljit_uw)code + put_label->label->size >= put_label->label->size`

Reply:
only if we are assuming the compiler doesn't optimize it out

Reply:
I have worked on ECMAScript BigInts recently, and v8 js engine heavily rely on similar checks to detect carry, and I also have no trouble with it so far.